db/snapshot/backup_task.cc

```diff
@@ -76,7 +78,7 @@ future<> backup_task_impl::upload_component(sstring name) {
     //  - s3::client::claim_memory semaphore
     //  - http::client::max_connections limitation
     try {
-        co_await _client->upload_file(component_name, destination, _progress, &_as);
+        co_await client->upload_file(component_name, destination, progress, &as);
     } catch (...) {
         snap_log.error("Error uploading {}: {}", component_name.native(), std::current_exception());
         throw;
```

Comment:
Shouldn't this be:
```suggestion
        throw std::current_exception();
```
Otherwise we propagate the exception itself (e.g. storage_io_error) and it is not caught as expected.